core/src/main/java/hudson/security/csrf/CrumbFilter.java

```diff
@@ -93,6 +87,21 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
         }
     }
 
+    private String extractCrumbFromRequest(HttpServletRequest httpRequest, String crumbFieldName) {
+        String crumb = httpRequest.getHeader(crumbFieldName);
+        if (crumb == null) {
```

Comment:
BTW clearer to `return` a value as soon as you find one, then end the method with `return null`.